library/core/src/macros/mod.rs

```diff
@@ -198,14 +198,11 @@ pub macro assert_matches {
 
 /// A macro for defining `#[cfg]` match-like statements.
 ///
-/// It is similar to the `if/elif` C preprocessor macro by allowing definition of a cascade of
-/// `#[cfg]` cases, emitting the implementation which matches first.
+/// The `cfg_select!` macro checks a series of `#[cfg]` conditions, and picks the branch that
+/// belongs to the first condition that evaluates to true.
 ///
-/// This allows you to conveniently provide a long list `#[cfg]`'d blocks of code
-/// without having to rewrite each clause multiple times.
-///
-/// Trailing `_` wildcard match arms are **optional** and they indicate a fallback branch when
-/// all previous declarations do not evaluate to true.
+/// An optional trailing `_` wildcard match arm can be used to specify a fallback branch.
+/// If none of the conditions evaluate to `true`, a compile error is emitted.
```

Comment:
```suggestion
/// An optional trailing `_` wildcard can be used to specify a fallback. If
/// none of the predicates are true, a [`compile_error`] is emitted.
```

I.e., avoid `match` language for now.  We will have to decide, for the Reference, whether we want to call these "arms", "branches", etc.